python/tskit/tables.py

```diff
@@ -585,6 +586,42 @@ def _update_metadata_schema_cache_from_ll(self) -> None:
             self.ll_table.metadata_schema
         )
 
+    def metadata_vector(self, key, *, dtype=None, default_value=None):
+        """
+        Returns a numpy array of metadata values obtained by extracting ``key``
+        from each metadata entry, and inserting ``default_value`` if the key is
+        not present. ``key`` may be a list, in which case nested values are returned.
+        For instance, ``key = ["a", "x"]`` will return an array of
+        ``row.metadata["a"]["x"]`` values, iterated over rows in this table.
+
+        :param str key: The name, or a list of names, of metadata entries.
+        :param dtype dtype: The dtype of the result (can usually be omitted).
+        :param object default_value: The value to be inserted if the metadata key
+            is not present. Note that for numeric columns, a default value of None
+            will result in a non-numeric array.
+        """
+
+        if isinstance(key, list):
+            out = np.array(
+                [
+                    functools.reduce(
+                        lambda d, k: d.get(k, default_value)
```

Comment:
We only want to use `get` if a default value is supplied so that `KeyError` is raised for a missing entry. As `None` could be a valid desired default, you'll need a singleton class `NO_DEFAULT` to check for and use as the default kwarg.

Reply:
oh dear